src/lib/eina/eina_value.c

```diff
@@ -1287,9 +1287,14 @@ _eina_value_type_int_convert_to(const Eina_Value_Type *type EINA_UNUSED, const E
             convert == EINA_VALUE_TYPE_STRING)
      {
         const char *other_mem;
-        char buf[64];
-        snprintf(buf, sizeof(buf), "%d", v);
-        other_mem = buf; /* required due &buf == buf */
+        if (type != EINA_VALUE_TYPE_ERROR)
+          {
+             char buf[64];
+             snprintf(buf, sizeof(buf), "%d", v);
+             other_mem = buf; /* required due &buf == buf */
+          }
+        else
+             other_mem = eina_error_msg_get(v);
```

Comment:
do not change `int_convert_to()`,  instead create a new `error_convert_to()` that will check conversion to string/stringshare, in that case use `eina_error_msg_get()`. Otherwise fallback to `int_convert_to()`